v2/fb-xcom-api-validator/src/server/middlewares/nonce-middleware.ts

```diff
@@ -0,0 +1,49 @@
+import logger from '../../logging';
+import { IncomingHttpHeaders } from 'http';
+import { FastifyReply, FastifyRequest, HookHandlerDoneFunction } from 'fastify';
+
+const log = logger('middleware:nonce');
+
+const usedNoncesForApiKeyMap = new Map<string, string[]>();
```

Comment:
Use map of [sets](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).